eureka_ml_insights/configs/aime.py

```diff
@@ -123,14 +127,132 @@ def configure_pipeline(
             ],
             output_dir=os.path.join(self.log_dir, "eval_report"),
         )
+        
+        # Aggregate the results by a majority vote 
+        self.postevalprocess_comp = EvalReportingConfig(
+            component_type=EvalReporting,
+            data_reader_config=DataSetConfig(
+                DataReader,
+                {
+                 "path": os.path.join(self.evalreporting_comp.output_dir, "metric_results.jsonl"),
+                 "format": ".jsonl",
+                 "transform":RunPythonTransform("df = df.loc[(df.groupby('ID')['model_output'].transform(lambda x: x.mode().iloc[0] if not x.mode().empty else x.iloc[0]) == df['model_output']) | (df['model_output'].isna() & df.groupby('ID')['model_output'].transform(lambda x: x.mode().iloc[0] if not x.mode().empty else x.iloc[0]).isna())].drop_duplicates('ID')")  # compute the majority vote over the model_output. If model_output is null, return null.
+                },
+            ),
+            metric_config=MetricConfig(ExactMatch),
+            aggregator_configs=[
+                AggregatorConfig(
+                    CountAggregator,
+                    {
+                        "column_names": [
+                            "ExactMatch_result",
+                        ],
+                        "group_by": "Year",
+                        "filename_base": "ExactMatch_GroupBy",
+                    },
+                ),
+            ],
+            output_dir=os.path.join(self.log_dir, "eval_report_majorityVote"),
+        )       
 
         # Configure the pipeline
         return PipelineConfig(
             [
                 self.data_processing_comp,
                 self.inference_comp,
                 self.data_post_processing,
+           
                 self.evalreporting_comp,
+                self.postevalprocess_comp,
             ],
             self.log_dir,
         )
+
+
+class AIME_PIPELINE5Run(AIME_PIPELINE):
+    """This class specifies the config for running AIME benchmark 5 repeated times"""
+    def configure_pipeline(
+        self, model_config: ModelConfig, resume_from: str = None, **kwargs: dict[str, Any]
+    ) -> PipelineConfig:
+        super().configure_pipeline(model_config=model_config, resume_from=resume_from)
+        # data preprocessing
+        self.data_processing_comp.data_reader_config.init_args["transform"]  = SequenceTransform(
+                        [
+                            ColumnRename(
+                                name_mapping={
+                                    "Question": "prompt",
+                                    "Answer": "ground_truth",
+                                }
+                            ),
+                            MultiplyTransform(n_repeats=5),                            

```

Comment:
you can append this to the transforms to avoid repeating the previous transforms in the sequence

Reply:
fixed